mathesar_ui/src/component-library/radio-group/RadioGroup.svelte

```diff
@@ -0,0 +1,27 @@
+<script lang="ts">
+  import type { RadioGroupOption } from '@mathesar-component-library/types';
+
+  export let group: string;
+  export let idKey = 'id';
+  export let labelKey = 'label';
+  export let isInline = true;
+  export let options: RadioGroupOption[] = [];
+</script>
+
+<fieldset class="radio-group">
+  <legend>
+    <slot></slot>
+  </legend>
+  <div class:inline={isInline}>
+    {#each options as option (option[idKey])}
+      <label class="radio">
+          <input type="radio" bind:group value={option[idKey]}>
```

Comment:
TypeScript is complaining on this line. I'm fine if we address this in a subsequent PR. I'd like an opinion from @pavish on the best approach. 

One fix is to change `RadioGroup.d.ts` to:

```ts
export interface RadioGroupOption {
  [key: string]: string | number | string[]
}
```

I know you were following the pattern that Pavish set in `Select.d.ts`. In the context of the RadioGroup component, I can't immediately see the usefulness of making `RadioGroupOption` generic. Pavish probably had a good reason to make `SelectOption` generic, but I'm not sure if that applies here. 

I would have designed these components with static keys for the label and id. My personal opinion is that it would simplify things a bit, especially the typing.

Reply:
I'd change it to 
```
export interface RadioGroupOption<T = unknown> {
  [key: string]: T
}
```

This will keep it generic, for cases where we need to extend from this, and the default would be unknown.

Reply:
For radio value, I think we can go with `string | number | string[]` since these are the only options that can be used as value for the input element.

`SelectOption` was made generic because the entire Select component is a custom element which can use objects as id or value.